writeaheadlog.go

```diff
@@ -411,6 +411,10 @@ func unmarshalPage(p *page, b []byte) (nextPage uint64, err error) {
 	// Read payloadSize
 	var payloadSize uint64
 	err5 := binary.Read(buffer, binary.LittleEndian, &payloadSize)
+	if payloadSize == 0 || payloadSize > maxPayloadSize {
+		err = errors.New("invalid page payload size")
+		return
+	}
```

Comment:
is `0` really invalid? Probably, but want @DavidVorick to confirm

Reply:
Probably. If the payload size is `0` then the page shouldn't have been allocated in the first place. I guess the first page could have payload size `0` but this would mean that the transaction is empty and we don't allow empty transactions in `NewTransaction`